python/llm/src/ipex_llm/transformers/models/qwen_vl.py

```diff
@@ -139,28 +174,42 @@ def qwen_attention_forward_vl(
                                                          device=hidden_states.device)
         new_key_states[:] = key.transpose(1, 2)
         new_value_states[:] = value.transpose(1, 2)
-        key = new_key_states.transpose(1, 2)
-        value = new_value_states.transpose(1, 2)
+        key = new_key_states
+        value = new_value_states
 
     if use_cache:
-        present = (key, value)
+        present = (key.transpose(1, 2), value.transpose(1, 2))
     else:
         present = None
 
+    if decoding_fast_path:
+        query = query.transpose(1, 2) # change to (bsz, q_len, num_heads, head_dim)
+
     if self.use_logn_attn and not self.training:
         if self.logn_tensor.device != query.device or self.logn_tensor.dtype != query.dtype:
             self.logn_tensor = self.logn_tensor.to(query.device).type_as(query)
-        seq_start = key.size(1) - query.size(1)
-        seq_end = key.size(1)
+        seq_start = key_size - key_size
```

Comment:
please check if the `seq_start` logic is correct.

Reply:
already tested this logic